tests/Doctrine/Tests/DBAL/Functional/Types/DecimalTest.php

```diff
@@ -32,4 +32,27 @@ public function testInsertAndRetrieveDecimal(): void
 
         self::assertSame('13.37', $value);
     }
+
+    public function testInsertAndRetrieveDecimalInteger(): void
+    {
+        $table = new Table('decimal_table');
+        $table->addColumn('val', Types::DECIMAL, ['precision' => 4, 'scale' => 2]);
+
+        $sm = $this->connection->getSchemaManager();
+        $sm->dropAndCreateTable($table);
+
+        $this->connection->insert(
+            'decimal_table',
+            ['val' => '13.0'],
+            ['val' => Types::DECIMAL]
+        );
+
+        $value = Type::getType(Types::DECIMAL)->convertToPHPValue(
+            $this->connection->fetchOne('SELECT val FROM decimal_table'),
+            $this->connection->getDatabasePlatform()
+        );
+
+        self::assertSame('13', $value);
```

Comment:
This is too restrictive. Implementations that behave correctly (like MySQL and Postgres) will return a string with as many decimals as you requested via `scale`, in this case `'13.00'`.

Reply:
Hm, I can't format it back with sqlite 🤔 So maybe striping trailing zeros in test would be sufficient?